src/device/rsp/rsp_core.c

```diff
@@ -321,16 +326,26 @@ void do_SP_Task(struct rsp_core* sp)
         if (sp->r4300->mi.regs[MI_INTR_REG] & MI_INTR_SP) {
             add_interrupt_event(&sp->r4300->cp0, SP_INT, 0/*100*/);
         }
-        sp->r4300->mi.regs[MI_INTR_REG] &= ~MI_INTR_SP;
         sp->regs[SP_STATUS_REG] &= ~SP_STATUS_TASKDONE;
     }
+
+    if ((sp->regs[SP_STATUS_REG] & SP_STATUS_HALT) == 0 && sp->r4300->mi.regs[MI_INTR_REG] == 0) {
+        repeat = 1;
+        add_interrupt_event(&sp->r4300->cp0, SP_INT, 0x200);
+    } else {
+        add_interrupt_event(&sp->r4300->cp0, SP_INT, 0x200);
```

Comment:
I don't think this is right, it looks like Project64 only add an interrupt when it has to repeat.